scalikejdbc-core/src/main/scala/scalikejdbc/ParameterBinder.scala

```diff
@@ -15,20 +15,27 @@ import java.sql.PreparedStatement
  * sql"insert into table (bin) values (${bin})".update.apply()
  * }}}
  */
-trait ParameterBinder[A] {
-
-  /**
-   * Parameter value.
-   */
-  @deprecated("This unused field will be removed", since = "2.2.4")
-  def value: A
+trait ParameterBinder { self =>
 
   /**
    * Applies parameter to PreparedStatement.
    */
   def apply(stmt: PreparedStatement, idx: Int): Unit
 
 }
+trait ParameterBinderWithValue[A] extends ParameterBinder { self =>
+
+  def value: A
+
+  // [[scalikejdbc.ParameterBinder#map]] breaks the Functor-law
```

Comment:
this method is dirty hack for logging/ParameterBinderFactory#xmap.
And it breaks the Functor-law.

so I dont think that it should be public to ScalikeJDBC users.